source/common/conn_pool/conn_pool_base.cc

```diff
@@ -393,6 +432,9 @@ void ConnPoolImplBase::drainConnectionsImpl(DrainBehavior drain_behavior) {
   // so use a for-loop since the list is not mutated.
   ASSERT(&owningList(ActiveClient::State::Draining) == &busy_clients_);
   for (auto& busy_client : busy_clients_) {
+    if (busy_client->state() == ActiveClient::State::Draining) {
+      continue;
```

Comment:
Can you explain this change? Is this a bug fix, or is it required by the new codepath?

Reply:
It's an unrelated change, but draining twice a client is no-op in our implementation. I could revert it though.